routingtable/filter/term_condition.go

```diff
@@ -18,7 +19,7 @@ func NewTermCondition(prefixLists []*PrefixList, routeFilters []*RouteFilter) *T
 }
 
 func (f *TermCondition) Matches(p net.Prefix, pa *route.Path) bool {
-	return f.matchesAnyPrefixList(p) || f.machtchesAnyRouteFilter(p)
+	return f.matchesAnyPrefixList(p) || f.machtchesAnyRouteFilter(p) || f.machtchesAnyLageCommunityFilter(pa)
```

Comment:
Regarding code maintenance I would prefer a TermCondition Type for every TermCondition and introduce a Interface for a TermCondition (prefixList, routeFilters, largeCommunity). This makes it easier to add new TermConditions. However we should test it performance wise.

Reply:
Fixed the unneccessary cast. For refactoring of Matches(net.Prefix, *route.Path) function I will create a new feature branch in which we benchmark performance.